proto/p4/config/v1/p4info.proto

```diff
@@ -265,15 +265,26 @@ message ActionProfile {
   repeated uint32 table_ids = 2;
   // true iff the action profile used dynamic selection
   bool with_selector = 3;
-  // max total weight of all member entries across all groups, or max number of
-  // member entries across all groups if the action profile does not have a
-  // selector
+  // max number of member entries across all groups if the action profile does
+  // not have a selector. Otherwise, semantics as specified by
+  // `selector_group_size_computation` below.
   int64 size = 4;
-  // max number of member entries across all groups
-  int64 max_num_members = 6;
-  // max number of weighted member entries in any given selector group, or 0 if
-  // the action profile does not have a selector
+  // 0 if the action profile does not have a selector. Otherwise, semantics as
+  // specified by `selector_group_size_computation` below.
   int32 max_group_size = 5;
+  // specifies the semantics of `size` and `max_group_size` above
+  message SumOfWeights {}
+  message SumOfMembers {}
+  oneof selector_group_size_computation {
```

Comment:
Any strong reasons why you went with a oneof here?
Perhaps this should be an enum, with default value `SUM_OF_WEIGHTS = 0` for backward compatability?

Reply:
Just poor memory. Should clearly be an enum. Will fix on Monday!